core/src/main/java/org/apache/iceberg/avro/PruneColumns.java

```diff
@@ -149,13 +150,14 @@ public Schema array(Schema array, Schema element) {
         Schema valueProjection = element.getField("value").schema();
         // it is possible that key is not selected, and
         // key schemas can be different if new field ids were assigned to them
-        if (keyProjectionField != null && keyValue.getField("key").schema() != keyProjectionField.schema()) {
+        if (keyProjectionField != null && !Objects.equals(keyValue.getField("key").schema(),
+            keyProjectionField.schema())) {
```

Comment:
Minor: Try to wrap lines close to the top level, rather than wrapping inside of argument lists. Wrapping after `&&` makes this more readable.

Reply:
my bad as I didn't apply code formatting after adjusting the code.

Reply:
pushed a new version